# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/credentialmanagement/CredentialsContainer.cpp
# Commit: 287bedb227e8
# Full Hash: 287bedb227e89933347633242efe1acd4236ad70
# Author: John Schanck <jschanck@mozilla.com>
# Date: 2025-02-12 09:32:07
# Regressor Bug: 1945969
# File Overlap Count: 1
# Description:
#   Bug 1945969 - part 5: rename WebAuthnManager to WebAuthnHandler. r=keeler
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D237257
# ==============================================================================

diff -r a01c165a2524 -r 287bedb227e8 dom/credentialmanagement/CredentialsContainer.cpp
--- a/dom/credentialmanagement/CredentialsContainer.cpp	Wed Feb 12 02:12:58 2025 +0000
+++ b/dom/credentialmanagement/CredentialsContainer.cpp	Wed Feb 12 02:12:58 2025 +0000
@@ -14,7 +14,7 @@
 #include "mozilla/dom/Promise-inl.h"
 #include "mozilla/StaticPrefs_dom.h"
 #include "mozilla/StaticPrefs_security.h"
-#include "mozilla/dom/WebAuthnManager.h"
+#include "mozilla/dom/WebAuthnHandler.h"
 #include "mozilla/dom/WindowGlobalChild.h"
 #include "mozilla/dom/WindowContext.h"
 #include "nsContentUtils.h"
@@ -24,7 +24,8 @@
 
 namespace mozilla::dom {
 
-NS_IMPL_CYCLE_COLLECTION_WRAPPERCACHE(CredentialsContainer, mParent, mManager)
+NS_IMPL_CYCLE_COLLECTION_WRAPPERCACHE(CredentialsContainer, mParent,
+                                      mWebAuthnHandler)
 NS_IMPL_CYCLE_COLLECTING_ADDREF(CredentialsContainer)
 NS_IMPL_CYCLE_COLLECTING_RELEASE(CredentialsContainer)
 NS_INTERFACE_MAP_BEGIN_CYCLE_COLLECTION(CredentialsContainer)
@@ -133,19 +134,19 @@
 
 CredentialsContainer::~CredentialsContainer() = default;
 
-void CredentialsContainer::EnsureWebAuthnManager() {
+void CredentialsContainer::EnsureWebAuthnHandler() {
   MOZ_ASSERT(NS_IsMainThread());
 
-  if (!mManager) {
-    mManager = new WebAuthnManager(mParent);
+  if (!mWebAuthnHandler) {
+    mWebAuthnHandler = new WebAuthnHandler(mParent);
   }
 }
 
-already_AddRefed<WebAuthnManager> CredentialsContainer::GetWebAuthnManager() {
+already_AddRefed<WebAuthnHandler> CredentialsContainer::GetWebAuthnHandler() {
   MOZ_ASSERT(NS_IsMainThread());
 
-  EnsureWebAuthnManager();
-  RefPtr<WebAuthnManager> ref = mManager;
+  EnsureWebAuthnHandler();
+  RefPtr<WebAuthnHandler> ref = mWebAuthnHandler;
   return ref.forget();
 }
 
@@ -191,9 +192,10 @@
       return promise.forget();
     }
 
-    EnsureWebAuthnManager();
-    return mManager->GetAssertion(aOptions.mPublicKey.Value(),
-                                  conditionallyMediated, aOptions.mSignal, aRv);
+    EnsureWebAuthnHandler();
+    return mWebAuthnHandler->GetAssertion(aOptions.mPublicKey.Value(),
+                                          conditionallyMediated,
+                                          aOptions.mSignal, aRv);
   }
 
   if (aOptions.mIdentity.WasPassed() &&
@@ -265,9 +267,9 @@
       return CreateAndRejectWithNotAllowed(mParent, aRv);
     }
 
-    EnsureWebAuthnManager();
-    return mManager->MakeCredential(aOptions.mPublicKey.Value(),
-                                    aOptions.mSignal, aRv);
+    EnsureWebAuthnHandler();
+    return mWebAuthnHandler->MakeCredential(aOptions.mPublicKey.Value(),
+                                            aOptions.mSignal, aRv);
   }
 
   if (aOptions.mIdentity.WasPassed() &&
@@ -304,8 +306,8 @@
       return CreateAndRejectWithNotAllowed(mParent, aRv);
     }
 
-    EnsureWebAuthnManager();
-    return mManager->Store(aCredential, aRv);
+    EnsureWebAuthnHandler();
+    return mWebAuthnHandler->Store(aCredential, aRv);
   }
 
   if (type.EqualsLiteral("identity") &&